storage/libradb/src/libradb_test.rs

```diff
@@ -489,3 +494,52 @@ proptest! {
         assert!(events.is_empty());
     }
 }
+
+#[test]
+fn test_get_latest_tree_state() {
+    let tmp_dir = TempPath::new();
+    let db = LibraDB::new(&tmp_dir);
+
+    // entirely emtpy db
+    let empty = db.get_latest_tree_state().unwrap();
+    assert_eq!(
+        empty,
+        TreeState::new(0, vec![], *SPARSE_MERKLE_PLACEHOLDER_HASH,)
+    );
+
+    // unbootstrapped db with pre-genesis state
+    let address = AccountAddress::new([0; 16]);
+    let blob = AccountStateBlob::from(vec![1]);
+    db.db
+        .put::<JellyfishMerkleNodeSchema>(
+            &NodeKey::new_empty_path(PRE_GENESIS_VERSION),
+            &Node::new_leaf(address.hash(), blob.clone()),
+        )
+        .unwrap();
```

Comment:
Is there a state store method that does the same thing, instead of writing raw key-value pair?

Reply:
no, because in TreeCache 1. we need a hack to set the node_cache up as a Node::Null at the root, at version PRE_GENESIS_VERSION, just like for empty db we need a virutal Node::Null at version 0; 2. after freezing, we tree_cache.next_version += 1, which breaks for PRE_GENESIS_VERSION == u64::max();

I spent much time to get around the above today until I realized it actually makes sense not to support that: in our imagined use case, the pre-genesis state should come from a back up, not through LibraDB APIs. Not providing such APIs makes things stricter hence safer.